test/unit/storage/entities/block.js

```diff
@@ -0,0 +1,669 @@
+/* eslint-disable mocha/no-pending-tests */
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { BaseEntity, Block } = require('../../../../storage/entities');
+const {
+	NonSupportedFilterTypeError,
+	NonSupportedOptionError,
+	NonSupportedOperationError,
+} = require('../../../../storage/errors');
+
+describe('Block', () => {
+	it('should be a constructable function', done => {
+		expect(Block.prototype.constructor).to.be.not.null;
+		expect(Block.prototype.constructor.name).to.be.eql('Block');
+		done();
+	});
+
+	it('should be be inherited by BaseEntity', done => {
+		expect(Block.prototype).to.be.an.instanceof(BaseEntity);
+		done();
+	});
+
+	describe('constructor()', () => {
+		it('should have one mandatory parameter', done => {
+			expect(Block.prototype.constructor.length).to.be.eq(1);
+			done();
+		});
+
+		it('should call super');
+
+		it('should assign proper parameters', done => {
+			const adapter = {
+				loadSQLFile: sinonSandbox.stub().returns('loadSQLFile'),
+			};
+			const defaultFilters = { version: 1 };
+
+			const block = new Block(adapter, defaultFilters);
+
+			expect(block.adapter).to.be.eq(adapter);
+			expect(block.defaultFilters).to.be.eq(defaultFilters);
+			expect(block.SQLs).to.be.eql({
+				selectSimple: 'loadSQLFile',
+				create: 'loadSQLFile',
+				isPersisted: 'loadSQLFile',
+			});
+			expect(block.FIELD_SET_SIMPLE.toString()).to.be.eql(
+				'Symbol(FIELD_SET_SIMPLE)'
+			);
+			expect(block.defaultOptions).to.be.eql({
+				fieldSet: block.FIELD_SET_SIMPLE,
+				limit: 10,
+				offset: 0,
+			});
+
+			done();
+		});
+
+		it('should setup specific fields', done => {
+			const adapter = { loadSQLFile: sinonSandbox.stub() };
+			const block = new Block(adapter);
+
+			expect(block.fields).to.have.all.keys([
+				'blockSignature',
+				'generatorPublicKey',
+				'height',
+				'id',
+				'numberOfTransactions',
+				'payloadHash',
+				'payloadLength',
+				'previousBlockId',
+				'reward',
+				'timestamp',
+				'totalAmount',
+				'totalFee',
+				'version',
+			]);
+
+			done();
+		});
+
+		it('should setup specific filters', done => {
+			const adapter = { loadSQLFile: sinonSandbox.stub() };
+			const block = new Block(adapter);
+
+			expect(block.filters).to.have.all.keys([
```

Comment:
What if we move this to a mocks/ directory? In pears I'm doing the same test and the test would be cleaner if required()

Reply:
Agreed.
@yatki is working on a proposal for how to structure the tests.
Can you give your thoughts here, @yatki ?